std/path.d

```diff
@@ -1725,21 +1725,22 @@ if (Ranges.length >= 2 &&
     Returns: The assembled path.
 */
 immutable(C)[] buildNormalizedPath(C)(const(C[])[] paths...)
-    @trusted pure nothrow
+    @safe pure nothrow
 if (isSomeChar!C)
 {
     import std.array : array;
+    import std.exception : assumeUnique;
 
-    const(C)[] result;
+    const(C)[] chained;
     foreach (path; paths)
     {
-        if (result)
-            result = chainPath(result, path).array;
+        if (chained)
+            chained = chainPath(chained, path).array;
         else
-            result = path;
+            chained = path;
     }
-    result = asNormalizedPath(result).array;
-    return cast(typeof(return)) result;
+    auto result = asNormalizedPath(chained).array;  // .array returns a copy, so we know it is unique
+    return () @trusted { return assumeUnique(result); } ();
```

Comment:
I realize that this has become an accepted way to do `@trusted`. But I think it actually goes against the idea behind it.

The `@trusted` snippet doesn't ensure safety by itself. It relies on the rest of the code not messing with `result`. That means, you can trigger undefined behavior by editing `@safe` code. Exactly what should not be possible.

Reply:
Good point.